packages/trackerless-network/src/logic/NodeList.ts

```diff
@@ -40,13 +41,18 @@ export class NodeList extends EventEmitter<Events> {
     }
 
     remove(nodeId: DhtAddress): void {
-        this.nodes.delete(nodeId)
+        if (this.nodes.has(nodeId)) {
+            const remote = this.nodes.get(nodeId)!
+            this.nodes.delete(nodeId)
+            this.emit('nodeRemoved', nodeId, remote)
+        }   
     }
 
     has(nodeId: DhtAddress): boolean {
         return this.nodes.has(nodeId)
     }
 
+    // Replace nodes does not emit nodeRemoved events, use with caution
```

Comment:
Maybe it could emit the events? Or could add a parameter which controls the event emitting.

Reply:
Yeah a parameter could make sense. The method is never called for the neighbor structure, only for contacts. Contact structures don't need the event for anything at the moment